src/scrapers/leumi.ts

```diff
@@ -166,14 +205,49 @@ async function fetchTransactions(page: Page, startDate: Moment): Promise<Transac
     throw new Error('Failed to extract or parse the account number');
   }
 
-  for (const accountId of accountsIds) {
-    if (accountsIds.length > 1) {
-      // get list of accounts and check accountId
-      await clickByXPath(page, 'xpath///*[contains(@class, "number") and contains(@class, "combo-inner")]');
-      await clickByXPath(page, `xpath///span[contains(text(), '${accountId}')]`);
+  debug(`Found ${accountsIds.length} account(s)`);
+
+  // When users have multiple accounts, Leumi displays the LAST account in the dropdown list by default.
+  // We need to:
+  // 1. Fetch the last account WITHOUT AccountIndex modification (gets the default displayed data)
+  // 2. Fetch remaining accounts using AccountIndex starting from 1
+  //
+  // For 2 accounts specifically:
+  // - accountsIds[1] (second/last in dropdown): fetch without AccountIndex
+  // - accountsIds[0] (first in dropdown): fetch with AccountIndex 1
+
+  // Create a mapping array with account info and the AccountIndex to use
+  const accountsToFetch =
+    accountsIds.length === 2
+      ? [
+          { id: accountsIds[1], accountIndex: undefined }, // Second in dropdown, no AccountIndex (default)
+          { id: accountsIds[0], accountIndex: 1 }, // First in dropdown, AccountIndex 1
+        ]
+      : accountsIds.map((id, i) => ({ id, accountIndex: i === 0 ? undefined : i }));
+
+  for (let i = 0; i < accountsToFetch.length; i++) {
+    const { id: accountId, accountIndex } = accountsToFetch[i];
+    try {
+      debug(
+        `Processing account ${i + 1}/${accountsToFetch.length}: ${accountId}${
+          accountIndex !== undefined ? ` with AccountIndex ${accountIndex}` : ' (no AccountIndex)'
+        }`,
+      );
```

Comment:
Consider the following:

```suggestion
  const accountsToFetch =
    accountsIds.length === 2
      ? [
          { accountId: accountsIds[1], i: 0, accountIndex: undefined }, // Second in dropdown, no AccountIndex (default)
          { accountId: accountsIds[0], i: 1, accountIndex: 1 }, // First in dropdown, AccountIndex 1
        ]
      : accountsIds.map((accountId, i) => ({ accountId, i, accountIndex: i === 0 ? undefined : i }));

  for (const { accountId, accountIndex, i } of accountsToFetch) {
    try {
      debug(`Processing account ${i + 1}/${accountsToFetch.length}:`, { accountId, accountIndex });
```